webapp/channels/src/components/channel_header_dropdown/menu_items/view_pinned_posts/view_pinned_posts.tsx

```diff
@@ -19,33 +20,32 @@ type Props = {
     };
 }
 
-export default class ViewPinnedPosts extends React.PureComponent<Props> {
-    private handleClick = (e: React.MouseEvent) => {
+const ViewPinnedPosts = ({
+    channel,
+    hasPinnedPosts,
+    actions: {
+        closeRightHandSide,
+        showPinnedPosts,
+    },
+    show,
+}: Props) => {
+    const handleClick = useCallback((e: MouseEvent) => {
         e.preventDefault();
 
-        const {
-            channel,
-            hasPinnedPosts,
-            actions: {
-                closeRightHandSide,
-                showPinnedPosts,
-            },
-        } = this.props;
-
         if (hasPinnedPosts) {
             closeRightHandSide();
         } else {
             showPinnedPosts(channel.id);
         }
-    };
-
-    render() {
-        return (
-            <Menu.ItemAction
-                show={this.props.show}
-                onClick={this.handleClick}
-                text={localizeMessage('navbar.viewPinnedPosts', 'View Pinned Posts')}
-            />
-        );
-    }
-}
+    }, [channel.id, closeRightHandSide, showPinnedPosts, hasPinnedPosts]);
+
+    return (
+        <Menu.ItemAction
+            show={show}
+            onClick={handleClick}
+            text={localizeMessage('navbar.viewPinnedPosts', 'View Pinned Posts')}
```

Comment:
Not related, but since we are here...

We are trying to get rid of the `localizeMessage` function. Could you use here intl directly?

What I mean is get the intl object from the hook at the beginning of the component (before instantiating the callback) (i.e.
```
const ViewPinnedPosts = ({...}: Props) => {
    const intl = useIntl();
    const handleClick = ...
```
) and then use `intl.formatMessage` to localize this message (i.e. `text={intl.formatMessage({id: 'navbar.viewPinnedPosts', defaultMessage: 'View Pinned Posts')}`).

Thanks!

Reply:
Done. 🙌